helix-view/src/view.rs

```diff
@@ -402,9 +404,53 @@ impl View {
         Some(pos)
     }
 
+    /// Get the text annotations to display in the current view for the given document and theme.
     pub fn text_annotations(&self, doc: &Document, theme: Option<&Theme>) -> TextAnnotations {
         // TODO custom annotations for custom views like side by side diffs
-        doc.text_annotations(theme)
+
+        let mut text_annotations = doc.text_annotations(theme);
+
+        let DocumentInlayHints {
+            id: _,
+            type_inlay_hints,
+            parameter_inlay_hints,
+            other_inlay_hints,
+            padding_before_inlay_hints,
+            padding_after_inlay_hints,
+        } = match doc.inlay_hints.get(&self.id) {
+            Some(doc_inlay_hints) => doc_inlay_hints,
+            None => return text_annotations,
+        };
+
+        let type_style = theme
+            .and_then(|t| t.find_scope_index("ui.virtual.inlay-hint.type"))
+            .map(Highlight);
+        let parameter_style = theme
+            .and_then(|t| t.find_scope_index("ui.virtual.inlay-hint.parameter"))
+            .map(Highlight);
+        let other_style = theme
+            .and_then(|t| t.find_scope_index("ui.virtual.inlay-hint"))
+            .map(Highlight);
+        let padding_style = theme
+            .and_then(|t| t.find_scope_index("ui.virtual.inlay-hint.padding"))
+            .map(Highlight);
```

Comment:
Ah one small think I forgot to add a review comment for:

I don't think this highlight scope should exist.
The LSP spec specifically states that the padding should have the same background color as the rest of the editor and NOT the same background as the virtual text.

Because of the way theme scopes work this will fallback to `ui.virtual.inlay-hint` if a theme doesn't define `ui.virtual.inlay-hint.padding` and therefore would not work as the LSP spec intended. I don't think it's necessary to style this empty space at all since the LSP spec is very specific that it should have the same background color as normal text.

You can simply use `None` here (in that case `ui.text` will be automatically used) :+1: 
```suggestion
        let padding_style = None;
```

Reply:
I don't agree with the spec here, I find it confusing in monospaced text to have a regular space I can't reach with my cursor, but done, you're right we should follow the spec regardless